Sources/SafeDICore/Generators/ScopeGenerator.swift

```diff
@@ -283,6 +283,24 @@ actor ScopeGenerator: CustomStringConvertible {
         }
     }
 
+    func generateDOT() async throws -> String {
+        let orderedPropertiesToGenerate = try orderedPropertiesToGenerate
+        let instantiatedProperties = orderedPropertiesToGenerate.map(\.scopeData.asDOTNode)
+        var childDOTs = [String]()
+        for orderedPropertyToGenerate in orderedPropertiesToGenerate {
+            let childDOT = try await orderedPropertyToGenerate.generateDOT()
+            if !childDOT.isEmpty {
+                childDOTs.append(childDOT)
+            }
+        }
+
+        let root = scopeData.asDOTNode
+        let forwardedProperties = scopeData.forwardedProperties.sorted().map { "\"\($0.asSource)\"" }
+        return ((instantiatedProperties + forwardedProperties).map {
+            "    \(root) -- \($0)"
```

Comment:
there's room for improvement here – ideally we'd write complete parent trees rather than just couplings to ensure better trees... but, one step at a time.

Reply:
just pushed up a commit with this.

Reply:
and reverted – it wasn't as good!